src/transformers/pipelines/question_answering.py

```diff
@@ -349,12 +345,18 @@ def preprocess(self, example, padding="do_not_pad", doc_stride=None, max_questio
             for k, v in feature.__dict__.items():
                 if k in model_input_names:
                     if self.framework == "tf":
-                        tensor = tf.constant(v)
+                        if isinstance(v, np.ndarray) and v.dtype == object:
+                            tensor = tf.constant(v, dtype=tf.int32)
+                        else:
+                            tensor = tf.constant(v)
```

Comment:
Not a big fan of this since `dtype == object` does not necessarily mean `dtype=tf.int32` logically (I know here it does make sense)